presto-main/src/test/java/com/facebook/presto/operator/aggregation/differentialentropy/TestExplicitWeightedReservoirAggregation.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.differentialentropy;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+import static com.facebook.presto.block.BlockAssertions.createRLEBlock;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static io.airlift.slice.Slices.wrappedBuffer;
+
+public class TestExplicitWeightedReservoirAggregation
```

Comment:
I see two version of differential_entropy function that appear to be doing exactly the same thing. One takes (size, sample, weight). Another takes (size, sample, weight, "reservoir_vasicek"). Why is it useful to have both functions?

Reply:
It personally makes sense to me, because when there is a method parameter, allowing to explicitly pass a value allows future-proofing in case another method (corresponding to a different implementation) is later added. Since it's tedious to always spell out parameters, there are shorter versions with default implementations (which are esp. convenient for interactive envs like Bento or Daiquery). This allows choice between less typing and clarity + future proofing, therefore.

WDYT?

Reply:
@atavory It is not clear to me what `future-proofing` means here. If we keep only one function without method parameter, we can always add another one with method parameter.